demos/ota/aws_iot_ota_update_demo.c

```diff
@@ -48,242 +50,257 @@
 #include "aws_demo_config.h"
 #include "aws_application_version.h"
 
-static void App_OTACompleteCallback( OTA_JobEvent_t eEvent );
-
-static BaseType_t prxCreateNetworkConnection( void );
-static IotNetworkError_t prvNetworkDisconnectCallback( void * pvContext );
-static void prvNetworkStateChangeCallback( uint32_t ulNetworkType,
-                                           AwsIotNetworkState_t xNetworkState,
-                                           void * pvContext );
-
-/*-----------------------------------------------------------*/
-
-#define otaDemoCONN_TIMEOUT_MS            ( 2000UL )
-
-#define otaDemoCONN_RETRY_INTERVAL_MS     ( 5000 )
-
-#define otaDemoCONN_RETRY_LIMIT           ( 100 )
-
-#define otaDemoKEEPALIVE_SECONDS          ( 1200 )
-
-#define myappONE_SECOND_DELAY_IN_TICKS    pdMS_TO_TICKS( 1000UL )
+/**
+ * @brief Timeout for MQTT connection, if the MQTT connection is not establihsed within
+ * this time, the connect function returns #IOT_MQTT_TIMEOUT
+ */
+#define OTA_DEMO_CONNECTION_TIMEOUT                  ( 2000UL )
 
-#define otaDemoNETWORK_TYPES              ( AWSIOT_NETWORK_TYPE_ALL )
+/**
+ * @brief The maximum time interval that is permitted to elapse between the point at
+ * which the MQTT client finishes transmitting one control Packet and the point it starts
+ * sending the next.In the absence of control packet a PINGREQ  is sent. The broker must
+ * disconnect a client that does not send a a message or a PINGREQ packet in one and a
+ * half times the keep alive interval.
+ */
+#define OTA_DEMO_KEEP_ALIVE_SECONDS                  ( 120UL )
 
+/**
+ * @brief The delay used in the main OTA Demo task loop to periodically output the OTA
+ * statistics like number of packets received, dropped, processed and queued per connection.
+ */
+#define OTA_DEMO_ONE_SECOND_DELAY                    pdMS_TO_TICKS( 1000UL )
 
 /**
- * @brief Structure which holds the context for an MQTT connection within Demo.
+ * @brief The base interval in seconds for retrying network connection.
  */
-static MqttConnectionContext_t xConnection =
-{
-    .pvNetworkConnection = NULL,
-    .ulNetworkType       = AWSIOT_NETWORK_TYPE_NONE,
-    .xNetworkInfo        = IOT_MQTT_NETWORK_INFO_INITIALIZER,
-    .xMqttConnection     = IOT_MQTT_CONNECTION_INITIALIZER,
-    .xDisconnectCallback = prvNetworkDisconnectCallback
-};
+#define OTA_DEMO_CONN_RETRY_BASE_INTERVAL_SECONDS    ( 4U )
 
 /**
- * @brief Network manager subscription callback.
+ * @brief The maximum interval in seconds for retrying network connection.
  */
+#define OTA_DEMO_CONN_RETRY_MAX_INTERVAL_SECONDS     ( 360U )
 
-static IotNetworkManagerSubscription_t xSubscriptionHandle = IOT_NETWORK_MANAGER_SUBSCRIPTION_INITIALIZER;
+/**
+ * @brief The longest client identifier that an MQTT server must accept (as defined
+ * by the MQTT 3.1.1 spec) is 23 characters. Add 1 to include the length of the NULL
+ * terminator.
+ */
+#define OTA_DEMO_CLIENT_IDENTIFIER_MAX_LENGTH        ( 24 )
```

Comment:
Shouldn't this be available in an MQTT header file somewhere?

Reply:
This is not defined in MQTT library , the MQTT  demo has its definition similar to what we are doing here.